Added documentation for Okteto Insights feature going live on 1.14 (#512)
okteto/docs: src/content/administration/okteto-insights.mdx

By default, this feature is disabled. To enable it you just need to set the helm setting [`insights.enabled`](/self-hosted/administration/configuration.mdx/#insights) to `true` and upgrade your Okteto instance.

## How to consume Okteto Insights' data

Once you enable the feature, Okteto will create a new endpoint accesible through `https://okteto.SUBDOMAIN/metrics` where data will be available in Prometheus format. This new endpoint is protected with a bearer token which is auto-generated by Okteto, or you can specify your own through the secret defined under this [helm setting](/self-hosted/administration/configuration.mdx/#insights).

Then, you can configure a scrapper to connect to this endpoint using the proper token. An example of a Prometheus scrapper configuration is:

```yaml
      - job_name: 'okteto_scrapper'
        scrape_interval: 5m

        scheme: http
        static_configs:
          - targets: ['okteto.<SUBDOMAIN>:443'] # replace <SUBDOMAIN> with your Okteto subdomain
        authorization:
          type: Bearer
          credentials: <bearer-token> # replace <bearer-token> with the token generated by Okteto
```

> > Be aware that Okteto does not provide a Prometheus instance or storage for this data, so you need to deploy your own Prometheus instance (or other collector platform) to consume this data and take care of counters and metrics restart with the functions Prometheus provides.

Reviewer: ```suggestion
:::caution
Okteto does **not** provide a Prometheus instance or storage for this data.  You must deploy your own Prometheus instance (or other collector platform) to consume this data and manage counters and metrics restart with the functions Prometheus provides.
:::
```

## What data is available

> Okteto will progressively add more data exposed through this endpoint, but the following metrics are available today:

Reviewer: ```suggestion
Okteto will progressively make more data available through this endpoint. The following metrics are available today:
```

> - `okteto_system_node`: This metric is a [gauge](https://prometheus.io/docs/concepts/metric_types/#gauge) which represents the number of nodes running Okteto. This metric doesn't have any label.

Reviewer: ```suggestion
- `okteto_system_node`: This metric is a [gauge](https://prometheus.io/docs/concepts/metric_types/#gauge) which represents the number of nodes running Okteto. This metric does not have any label.
```

@ifbyol just to confirm: "nodes running Okteto" -- I want to double check this wasn't intended to be "nodes running _in_ Okteto" or something similar. I'm assuming you meant what's written here, but I hadn't seen this phrasing before so thought I'd ask 😄

Author: I wanted to say: "This is the number of nodes in the Kubernetes cluster that are used to run Okteto and all the services and namespaces created/deployed through Okteto"

> - `okteto_system_node_memory_bytes`: This metric is a gauge which represents the total allocatable memory in bytes for a node. It has the following labels:

Reviewer: ```suggestion
- `okteto_system_node_memory_bytes`: This metric is a gauge which represents the total memory (in bytes) available for allocating to a node. It has the following labels:
```

    - `node_name`: It contains the node name.

> - `okteto_system_node_memory_requested_bytes`: This metric is a gauge which represents the total [requested](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/) memory in bytes for a node. It has the following labels:

Reviewer: ```suggestion
- `okteto_system_node_memory_requested_bytes`: This metric is a gauge which represents the total memory (in bytes) [requested](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/) for a node. It has the following labels:
```

    - `node_name`: It contains the node name.

> - `okteto_system_node_memory_usage_bytes`: This metric is a gauge which represents the total memory in use in bytes for a node. It has the following labels:

Reviewer: ```suggestion
- `okteto_system_node_memory_usage_bytes`: This metric is a gauge which represents the total memory (in bytes) in use for a node. It has the following labels:
```

    - `node_name`: It contains the node name.

> - `okteto_system_node_cpu_cores`: This metric is a gauge which represents the total allocatable CPU cores for a node. It has the following labels:

Reviewer: ```suggestion
- `okteto_system_node_cpu_cores`: This metric is a gauge which represents the total CPU cores available for allocating to a node. It has the following labels:
```

    - `node_name`: It contains the node name.

> - `okteto_system_node_cpu_requested_cores`: This metric is a gauge which represents the total [requested](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/) CPU cores for a node. It has the following labels:

Reviewer: ```suggestion
- `okteto_system_node_cpu_requested_cores`: This metric is a gauge which represents the total CPU cores [requested](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/) for a node. It has the following labels:
```

    - `node_name`: It contains the node name.
- `okteto_system_node_cpu_usage_cores`: This metric is a gauge which represents the total CPU cores in use for a node. It has the following labels:
    - `node_name`: It contains the node name.

> > All metrics provided by Okteto will have the prefix `okteto_`, so they are easily identifiable and filtered.

Reviewer: ```suggestion
> All metrics provided by Okteto will have the prefix `okteto_`
```

---
title: Okteto Insights
description: This page provides information about Okteto Insights feature.
sidebar_label: Okteto Insights
id: okteto-insights
---

> Okteto inherently generates a lot of data from developers using the platform. This includes product usage and behavior data (i.e. developer activities) and infrastructure utilization and performance data (i.e. infrastructure and platform activities). 

Reviewer: ```suggestion
Okteto inherently generates a lot of data from developers using the platform. This includes product usage and behavior data (i.e. developer activities) as well as infrastructure utilization and performance data (i.e. infrastructure and platform activities). 
```
